web-assembly/hosted/okta-hosted-login/okta-blazor-web-assembly-example/Program.cs

```diff
@@ -1,40 +1,45 @@
-using System;
-using System.Net.Http;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-using System.Text;
-using Microsoft.AspNetCore.Components.WebAssembly.Authentication;
-using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Logging;
-
-namespace okta_blazor_web_assembly_example.Client
-{
-    public class Program
-    {
-        public static async Task Main(string[] args)
-        {
-            var builder = WebAssemblyHostBuilder.CreateDefault(args);
-            builder.RootComponents.Add<App>("app");
-
-            builder.Services.AddHttpClient("okta_blazor_web_assembly_example.ServerAPI", client => client.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress))
-                .AddHttpMessageHandler<BaseAddressAuthorizationMessageHandler>();
-
-            // Supply HttpClient instances that include access tokens when making requests to the server project
-            builder.Services.AddScoped(sp => sp.GetRequiredService<IHttpClientFactory>().CreateClient("okta_blazor_web_assembly_example.ServerAPI"));
-
-            builder.Services.AddOidcAuthentication(options =>
-            {
-                // Replace the Okta placeholders with your Okta values in the appsettings.json file.
-                options.ProviderOptions.Authority = builder.Configuration.GetValue<string>("Okta:Authority");
-                options.ProviderOptions.ClientId = builder.Configuration.GetValue<string>("Okta:ClientId");;
-                options.ProviderOptions.ResponseType = "code";
-            });
-
-            builder.Services.AddApiAuthorization();
-
-            await builder.Build().RunAsync();
-        }
-    }
-}
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace okta_blazor_web_assembly_example
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var builder = WebAssemblyHostBuilder.CreateDefault(args);
+            builder.RootComponents.Add<App>("app");
+
+            //- For cross-domain requests to Api Server:
+            builder.Services.AddScoped<CorsRequestAuthorizationMessageHandler>(); 
+            builder.Services
+                .AddHttpClient("BlazorClient.ServerApi", client => client.BaseAddress = new Uri(builder.Configuration.GetValue<string>("ServerApi:BaseAddress")))
+                .AddHttpMessageHandler<CorsRequestAuthorizationMessageHandler>();
+            
+            //- Otherwise use the following instead:
+            //builder.Services

```

Comment:
Do we need this piece of code?

Reply:
Currently this client is configured to work with API server on another domain, but in my opinion the most probable use case is when API server is on the same domain as frontend site and in that case commented-out code can be used.

Reply:
I think it's better to show those two cases and corresponding customization options

Reply:
I see. Maybe we can provide more details in the comment, something like "If you don't need CORS, use the following piece of code instead."